modules/ephemeral-storage-additional-details-on-volumeattributes-on-shared-resource-pod-volumes.adoc

```diff
@@ -0,0 +1,39 @@
+:_content-type: REFERENCE
+
+[id="ephemeral-storage-additional-details-on-volumeattributes-on-shared-resource-pod-volumes_{context}"]
+= Additional details on VolumeAttributes on shared resource pod volumes
+
+[role="_abstract"]
+The following attributes determine affect shared resource pod volumes in various ways:
+
+* The `refreshResource` attribute in the `volumeAttributes` properties.
+* The `refreshResources` attribute in the Shared Resource CSI Driver configuration.
+* The `sharedSecret` and `sharedConfigMap` attributes in the `volumeAttributes` properties.
+
+== The `refreshResource` attribute
+
+The Shared Resource CSI Driver honors the `refreshResource` attribute in `volumeAttributes` properties of the volume. This attribute controls whether updates to the contents of the underlying `Secret` or `ConfigMap` object are copied to the volume *after* the volume is initially provisioned as part of pod startup. The default value of `refreshResource` is `true`, which means that contents are updated.
+
+[IMPORTANT]
+====
+The following `refreshResources` attribute overrides the `refreshResource` attribute in `volumeAttributes` properties of the volume.
+====
+
+== The `refreshResources` attribute
+
+You can use a global switch to enable or disable refreshing of shared resources. This switch is the `refreshResources` attribute in the `csi-driver-shared-resource-config` config map for the Shared Resource CSI Driver, which you can find in the `openshift-cluster-csi-drivers` namespace. If you set this `refreshResources` attribute to `false`, none of the `Secret` or `ConfigMap`-related content stored in the volume is updated after the initial provisioning of the volume.
+
+[IMPORTANT]
+====
+This `refreshResources` attribute overrides the `refreshResource` attribute in `volumeAttributes` properties of the volume.
+====
+
+== Validation of volumeAttributes before provisioning a shared resource volume for a pod
+
+In the `volumeAttributes` of a single volume, you must set either `sharedSecret` or `sharedConfigMap` attribute to the value of a `SharedSecret` or `SharedConfigMap` resource. Otherwise, during provisioning the volume during pod startup, a validation returns an error to the kubelet.
```

Comment:
```suggestion
In the `volumeAttributes` of a single volume, you must set either `sharedSecret` or `sharedConfigMap` attribute to the value of a `SharedSecret` or `SharedConfigMap` instance. Otherwise, during provisioning the volume during pod startup, the Shared Resource CSI driver returns an error to the kubelet.
```

Reply:
Updated. Thanks.